routes/markers.js

```diff
@@ -141,72 +141,80 @@ module.exports = function(connection) {
     /* Update marker with given id */
     router.patch('/marker/:id', utils.isAuthenticated, function (req, res, next) {
 
-        //If authenticated user is not the owner of the marker or an admin, return 401 (Unauthorized)
-        if(req.session.user.uid != req.body.uid && !req.session.user.admin) return res.send(401);
-
-        if(req.body !== null && !Object.keys(req.body).length) {
-            var id = req.params['id']
-            var date_changed=new Date().toISOString().slice(0, 19).replace('T', ' ');
-            var query = 'UPDATE atlas SET date_changed=? WHERE id =?';
-            if(!req.session.user.admin) {
-                query += ' AND created_by=\''+req.session.user.uid+'\'';
-            }
+        var id = req.params['id']
 
-            connection.query(query, [date_changed,id], function (error, rows,field) {
-                if(!!error){
-                    console.log(error);
-                }
-            });            
+        connection.query("SELECT created_by FROM atlas WHERE id=?", [id], function (error, rows, field) {
+            if(error) {
+                console.log(error);
+            }
+            else if(rows[0].created_by != req.session.user.uid && !req.session.user.admin) {
+                return res.send(401);
+            } else {
+                if(req.body !== null && !Object.keys(req.body).length) {
+                    var date_changed=new Date().toISOString().slice(0, 19).replace('T', ' ');
+                    var query = 'UPDATE atlas SET date_changed=? WHERE id =?';
+                    if(!req.session.user.admin) {
+                        query += ' AND created_by=\''+req.session.user.uid+'\'';
+                    }
+
+                    connection.query(query, [date_changed,id], function (error, rows,field) {
+                        if(!!error){
+                            console.log(error);
+                        }
+                    });            
+
+                    connection.query("SELECT * from atlas WHERE id=?", [id], function (error, rows,field) {
```

Comment:
Isn't the preceding query asynchronous (i.e., using a promise)? That would mean this line could execute before the UPDATE has completed, right? I believe this either needs to be nested our you could use `await connection.query(...)`.[[ref](https://medium.com/tkssharma/writing-neat-asynchronous-node-js-code-with-promises-async-await-fa8d8b0bcd7c)]

Reply:
Ah, missed that!